### PR TITLE
Fix urlFilterApplier for filter values as array

### LINKED ISSUE #1692: Remove adminhtm area emulation from media-content:sync command (if possible)
Remove adminhtm area emulation from media-content:sync command (if possible)

### LINKED ISSUE #1699: Empty "Used In" detail is displayed for images that are not used anywhere
### Preconditions

1. Install Magento with Adobe Stock Integration
2. Configured integration in Stores -> Configuration -> Advanced-> System -> Adobe Stock Integration fieldset
3. Enhanced Media Gallery Enabled
4. **Uploaded image which is not used anywhere**

### Steps to reproduce (*)
1. Select the uploaded image which is not Used anywhere
2. Click "three dots", select **View Details**
3. Observe the file Details

### Expected result (*)
If the image is not used anywhere, the _Used in_ Detail should not be displayed

### Actual result (*)
Empty _Used in_ is displayed
empty_used_in

### LINKED ISSUE #1702: Frontend/UI issues on adding new tag which contains maximum number of symbols, 128
### Description
The symbol numbers limit for new Tag is 128, add such tag and observe the behavior
To fix this issue, the styles, which are used in Asset filter, could be applied here.

### Preconditions

1. Install Magento with Adobe Stock Integration
2. Configured integration in Stores -> Configuration -> Advanced-> System -> Adobe Stock Integration fieldset
3. Enhanced Media Gallery Enabled
4. Uploaded image 

### Steps to reproduce (*)
1. Select the uploaded image by clicking on it
2. Click "three dots", select **Edit**
3. Place the cursor on _New Keyword_ field and type in 128 symbols
4. Click **Add New Tag**
5. Observe the **Tags** drop-down

### Actual result (*)
tags2

### LINKED ISSUE #1704: The asset can not be checked in Assets filter
### Preconditions

1. Install Magento with Adobe Stock Integration
2. Configured integration in Stores -> Configuration -> Advanced-> System -> Adobe Stock Integration fieldset
3. Enhanced Media Gallery Enabled
4. Uploaded image 

### Steps to reproduce (*)
1. Create a new Product and add the previously uploaded image to its Content
2. Go to **Content - Media Gallery**
3. Select the previously used image by clicking on it
4. Click "three dots", select **View Details**
5. The Used In Detail displays _Used in: Products(1)_ link. Click on _Products(1)_ link
(user is redirected to Products grid)
6. Click **Filters**
7. Dropdown **Asset** filter and type image name 
(image is found)
8. Try to check the image checkbox

### Expected result (*)
The image checkbox is checked, the image is added to Asset filter

### Actual result (*)
The checkbox is cannot be checked
Peek 2020-08-03 14-42

### LINKED ISSUE #1747: Used In secrion of asset details is formatted incorrectly if more than 2 entities referenced
### Steps to reproduce (*)
1. Add an image from media gallery to three different entities content (i.e. page, category and block)
2. View details of this image in the media gallery

### Expected result (*)
Used in entities should be listed one under another

### Actual result (*)
image

### PR DESCRIPTION
**Pull request by Nazar65**
### Related Pull Requests
https://github.com/magento/partners-magento2-infrastructure/pull/19

### Fixed Issues (if relevant)
'
2. Fixes magento/adobe-stock-integration#1704: The asset can not be checked in Assets filter
3. Fixes magento/adobe-stock-integration#1702: Frontend/UI issues on adding new tag which contains maximum number of symbols, 128 
4. Fixes magento/adobe-stock-integration#1692: Remove adminhtm area emulation from media-content:sync command (if possible) #1692
5. Fixes magento/adobe-stock-integration#1699: Empty "Used In" detail is displayed for images that are not used anywhere 
6. FIxes magento/adobe-stock-integration#1747 Used In secrion of asset details is formatted incorrectly if more than 2 entities referenced

### Manual testing scenarios (*)
. Install Magento with Adobe Stock Integration
2. Configured integration in Stores -> Configuration -> Advanced-> System -> Adobe Stock Integration fieldset
3. Enhanced Media Gallery Enabled
4. Uploaded image 

### Steps to reproduce (*)
1. Create a new Product and add the previously uploaded image to its Content
2. Go to **Content - Media Gallery**
3. Select the previously used image by clicking on it
4. Click "three dots", select **View Details**
5. The Used In Detail displays _Used in: Products(1)_ link. Click on _Products(1)_ link
(user is redirected to Products grid)
6. Click **Filters**
7. Dropdown **Asset** filter and type image name 
(image is found)
8. Try to check the image checkbox

### Expected result (*)
The image checkbox is checked, the image is added to Asset filter

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by rogyar** [comment]
@magento run Functional Tests B2B

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

File: app/code/Magento/Ui/view/base/web/js/grid/url-filter-applier.js [hunk 1]
```diff
@@ -75,6 +91,20 @@ define([
...
+        /**
+         * Get Filter options
+         *
+         * @param {String} url
+         */
+        getOptionsParam: function (url) {
+            var searchString = decodeURI(url),
+                options;
+            options = Object.fromEntries(new URLSearchParams(searchString));
+            delete options['filters[asset_id]'];
```
Comment (sivaschenko) [inline app/code/Magento/Ui/view/base/web/js/grid/url-filter-applier.js]: Please remove the `asset_id` hardcode

File: app/code/Magento/Ui/view/base/web/js/grid/url-filter-applier.js [hunk 2]
```diff
@@ -36,16 +40,23 @@ define([
...
+                !_.isNull(this.selectComponentName) && _.isUndefined(this.selectComponent())) {
                 setTimeout(function () {
                     this.apply();
                 }.bind(this), 100);
 
                 return;
             }
 
+            if (Object.keys(options).length) {
+                this.selectComponent().options(options);
```
Comment (sivaschenko) [inline app/code/Magento/Ui/view/base/web/js/grid/url-filter-applier.js]: I'd prefer to call `loadOptions` if the component is `ui-select` if possible

File: app/code/Magento/Ui/view/base/web/js/grid/url-filter-applier.js [hunk 3]
```diff
@@ -12,11 +12,15 @@ define([
     return Component.extend({
         defaults: {
             listingNamespace: null,
+            selectComponentName: null,
```
Comment (sivaschenko) [inline app/code/Magento/Ui/view/base/web/js/grid/url-filter-applier.js]: The filter may not be a select, please adjust the logic accordingly
```suggestion
            filterComponentName: null,
```

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the PR @Nazar65 ! Please see my comments

Also, considering the complexity - can you please extract the `url-filter-applier` changes to a separate pull request.

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Review by sivaschenko — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-7976](https://jira.corp.magento.com/browse/ENGCOM-7976) has been created to process this Pull Request

**Comment by sivaschenko** [comment]
@Nazar65 please create a PR with `ce-etalon.json` update in https://github.com/magento/partners-magento2-infrastructure and link both PRs to each other as related

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by chalov-anton** [comment]
:heavy_check_mark: QA Passed

- 1: 
 The Asset filter is applied. The asset name is displayed
Selection_061

- 2: 
The assets can be checked 
Peek 2020-08-07 17-53

- 3: 
the long tag name is shortened by three dots
long_tag_name

- 4:
`media-gallery:sync` and `media-content:sync` work correctly

- 5:
Used in is not displayed for images that are not used anywhere
used_in

All the cases are covered by Manual Testing Scenarios

**Comment by Nazar65** [comment]
@magento run all tests

**Review by sivaschenko — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-7976](https://jira.corp.magento.com/browse/ENGCOM-7976) has been created to process this Pull Request

**Pull request merged**